File: snippets/swift/Sources/Gossip.swift
import Foundation
// ANCHOR: gossip
import NostrSDK

func buildGossipClient(databasePath: String) async throws -> Client {
    let gossip = try await NostrGossip.sqlite(path: databasePath)
    let client = ClientBuilder().gossip(gossip: gossip).build()
    let capabilities = RelayCapabilities.discovery()

    _ = try await client.addRelay(
        url: try RelayUrl.parse(url: "wss://relay.damus.io"),
        capabilities: capabilities
    )
    _ = try await client.addRelay(
        url: try RelayUrl.parse(url: "wss://purplepag.es"),
        capabilities: capabilities
    )
    await client.connect()

    return client
}
// ANCHOR_END: gossip

File: snippets/js/react-native/src/gossip.ts
// ANCHOR: gossip
import {
    ClientBuilder,
    NostrGossip,
    RelayCapabilities,
    RelayUrl,
} from "@nostrdevkit/nostr-sdk-react-native";

export async function buildGossipClient(databasePath: string) {
    const gossip = await NostrGossip.sqlite(databasePath);
    const client = new ClientBuilder().gossip(gossip).build();
    const capabilities = RelayCapabilities.discovery();

    await client.addRelay(RelayUrl.parse("wss://relay.damus.io"), capabilities, false, undefined);
    await client.addRelay(RelayUrl.parse("wss://purplepag.es"), capabilities, false, undefined);
    await client.connect(undefined);

    return client;
}
// ANCHOR_END: gossip
